contentstore/tests/tests_server.py

```diff
@@ -0,0 +1,456 @@
+        resp = self.api.handle_request(
+            self.req_class('GET', path, None, self.headers))
+        return resp.code, resp.data
+
+    def post(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('POST', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def put(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('PUT', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def patch(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('PATCH', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def delete(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('DELETE', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def options(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('OPTIONS', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+
+class TestFakeContentStore(TestCase, ContentStoreApiTestMixin):
+
+    def setUp(self):
+        try:
+            from verified_fake.fake_contentstore import (Request,
+                                                         FakeContentStoreApi)
+        except ImportError as err:
+            if "verified_fake" not in err.args[0]:
+                raise
+            raise ImportError(" ".join([
+                err.args[0],
+                "(install from pypi or the 'verified_fake' directory)"]))
+
+        self.req_class = Request
+        self.api_class = FakeContentStoreApi
+        self.api = self.api_class("", "token-1", {}, {})
+        self.api.messagesets.endpoint_data = {}
+        self.api.schedules.endpoint_data = {}
+        self.api.messages.endpoint_data = {}
+        self.api.binary_contents.endpoint_data = {}
```

Comment:
Ah yes, this is caused by the default function values (`def __init__(self, data={})`). We could maybe do a `self.messagesets = {}`, and then create the api by `self.api = self.api_class(..., self.messagesets`. Then when creating/getting messagesets you can just do `self.messagesets[key] = value`. There's nothing wrong with doing it this way, I think it's just clearer doing it like that. I'm happy with both ways though.